packages/samples/headless-react/src/components/smart-snippet/smart-snippet.class.tsx

```diff
@@ -0,0 +1,104 @@
+import {Component, ContextType} from 'react';
+import {
+  buildSmartSnippet,
+  SmartSnippet as HeadlessSmartSnippet,
+  SmartSnippetState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {AppContext} from '../../context/engine';
+
+export class SmartSnippet extends Component<{}, SmartSnippetState> {
+  static contextType = AppContext;
+  context!: ContextType<typeof AppContext>;
+
+  private controller!: HeadlessSmartSnippet;
+  private unsubscribe: Unsubscribe = () => {};
+
+  componentDidMount() {
+    this.controller = buildSmartSnippet(this.context.engine!);
+    this.updateState();
+
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  private answerStyles(expanded: boolean) {
+    const maskImage = () =>
+      expanded
+        ? 'none'
+        : 'linear-gradient(to bottom, black 50%, transparent 100%)';
+
+    return {
+      maxHeight: expanded ? '100%' : '100px',
+      maxWidth: '200px',
+      overflow: 'hidden',
+      marginBottom: '10px',
+      maskImage: maskImage(),
+      WebkitMaskImage: maskImage(),
+    };
+  }
+
+  render() {
+    if (!this.state) {
+      return null;
+    }
+
+    const {
+      answerFound,
+      answer,
+      question,
+      liked,
+      disliked,
+      expanded,
+    } = this.state;
+
+    if (!answerFound) {
+      return <div>Sorry, no answer has been found for this query.</div>;
```

Comment:
Should we just return `null` in this case?

Reply:
For a sample purpose, I find it more meaningful to be more explicit about what that state mean.
 
That's a pattern we see in quite a lot of sample (eg: search status), that I find very useful.

I don't expect users to blindly copy paste that as is in their own app, but to get a general understanding of what it's about.